src/pf-dropdown/pf-dropdown.component.js

```diff
@@ -0,0 +1,172 @@
+/**
+ * <b>&lt;pf-dropdown&gt;</b> element for Patternfly Web Components
+ *
+ * <pf-dropdown id="dropdown1"><div class="dropdown">
+ *    <button class="btn btn-primary dropdown-toggle" type="button" id="dropdown2" data-toggle="dropdown" aria-haspopup="true"
+ *      aria-expanded="false">
+ *      Dropdown
+ *      <span class="caret"></span>
+ *      </button>
+ *      <ul class="dropdown-menu">
+ *        <li class="dropdown-header">Header</li>
+ *        <li><a href="#">Item 1</a></li>
+ *        <li><a href="#">Item 2</a></li>
+ *        <li class="disabled"><a href="#">Item 3</a></li>
+ *        <li><a href="#">Item 4</a></li>
+ *        <li class="divider"></li>
+ *        <li><a href="#">Item 5</a></li>
+ *      </ul>
+ *      </div>
+ *  </pf-dropdown>
+ */
+
+export class PfDropdown extends HTMLElement {
+
+  /**
+   * Called when an instance was inserted into the document
+   */
+  attachedCallback() {
+    this._button = this.querySelector('.btn');
+    this._disabled = /\bdisabled/.test(this._button.className);
+
+    this._button.addEventListener('click', () => {
+      this._showDropdown();
+    });
+
+    document.addEventListener('click', (event) => {
+      //close dropdown if clicked outside menu
+      if (event.target !== this && !this.contains(event.target)) {
+        this._clearDropdown();
+      }
+    });
+
+    document.addEventListener('keydown', (event) => {
+      if (/input|textarea/.test(event.target.tagName)) {
+        return;
+      }
+      if (this._disabled) {
+        return;
+      }
+      let keycode = event.keyCode ? event.keyCode : event.which;
+      let active = /\bopen/.test(this._button.parentNode.className);
+
+      //check if dropdown is open and keys other than 38,40,27,13 are not being used
+      if (active && /(38|40|27|13)/.test(keycode)) {
+        this._keyHandler(event, keycode);
```

Comment:
Not necessarily, I thought it would be good to check all keys before calling `keyHandler` method
if these keys are not pressed method will not be called. Nothing special.